skbio/io/format/embl.py

```diff
@@ -0,0 +1,1497 @@
+    return output
+
+
+# Method to determine if file is in EMBL format or not. A uniprot embl format
+# can't be parsed by this module (at the moment)
+@embl.sniffer()
+def _embl_sniffer(fh):
+    try:
+        line = next(_line_generator(fh, skip_blanks=True, strip=False))
+    except StopIteration:
+        return False, {}
+
+    try:
+        _parse_id([line])
+
+    except EMBLFormatError:
+        return False, {}
+
+    return True, {}
+
+
+@embl.reader(None)
+def _embl_to_generator(fh, constructor=None, **kwargs):
+    for record in _parse_embls(fh):
+        yield _construct(record, constructor, **kwargs)
+
+
+# Method to read EMBL data as skbio.sequence.DNA
+@embl.reader(Sequence)
+def _embl_to_sequence(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_embls(fh), seq_num)
+    return _construct(record, Sequence, **kwargs)
+
+
+# Method to read EMBL data as skbio.sequence.DNA
+@embl.reader(DNA)
+def _embl_to_dna(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_embls(fh), seq_num)
+    return _construct(record, DNA, **kwargs)
+
+
+# Method to read EMBL data as skbio.sequence.DNA
+@embl.reader(RNA)
+def _embl_to_rna(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_embls(fh), seq_num)
+    return _construct(record, RNA, **kwargs)
+
+
+# No protein support at the moment
+@embl.reader(Protein)
+def _embl_to_protein(fh, seq_num=1, **kwargs):
+    # no protein support, at the moment
+    # record = _get_nth_sequence(_parse_embls(fh), seq_num)
+    # return _construct(record, Protein, **kwargs)
+    raise EMBLFormatError("There's no protein support for EMBL record")
+
+
+# Writer methods
+@embl.writer(None)
+def _generator_to_embl(obj, fh):
+    for obj_i in obj:
+        _serialize_single_embl(obj_i, fh)
+
+
+@embl.writer(Sequence)
+def _sequence_to_embl(obj, fh):
+    _serialize_single_embl(obj, fh)
+
+
+@embl.writer(DNA)
+def _dna_to_embl(obj, fh):
+    _serialize_single_embl(obj, fh)
+
+
+@embl.writer(RNA)
+def _rna_to_embl(obj, fh):
+    _serialize_single_embl(obj, fh)
+
+
+@embl.writer(Protein)
+def _protein_to_embl(obj, fh):
+    # no protein support, at the moment
+    # _serialize_single_embl(obj, fh)
+    raise EMBLFormatError("There's no protein support for EMBL record")
+
+
+def _construct(record, constructor=None, **kwargs):
+    '''Construct the object of Sequence, DNA, RNA, or Protein.'''
+
+    # sequence, metadata and interval metadata
+    seq, md, imd = record
+
+    if 'lowercase' not in kwargs:
+        kwargs['lowercase'] = True
+
+    if constructor is None:
+        unit = md['LOCUS']['unit']
+        if unit == 'bp':
+            # RNA mol type has T instead of U for genbank from from NCBI
+            constructor = DNA
+
+        elif unit == 'aa':
+            # no protein support, at the moment
+            # constructor = Protein
+            raise EMBLFormatError("There's no protein support for EMBL record")
+
+    if constructor == RNA:
+        return DNA(
```

Comment:
Oh, I thought the EMBL records stored RNA as RNA? The skbio constructor does work assuming the sequence is actually RNA:

```python
In [1]: from skbio import RNA

In [2]: RNA('AUCG')
Out[2]:
RNA
--------------------------
Stats:
    length: 4
    has gaps: False
    has degenerates: False
    has definites: True
    GC-content: 50.00%
--------------------------
0 AUCG

In [3]: ?RNA

In [4]: RNA('ATCG')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-02895b4a2413> in <module>()
----> 1 RNA('ATCG')

/Users/mcdonadt/miniconda3/envs/sk050-biom215/lib/python3.5/site-packages/skbio/sequence/_grammared_sequence.py in __init__(self, sequence, metadata, positional_metadata, lowercase, validate)
    334
    335         if validate:
--> 336             self._validate()
    337
    338     def _validate(self):

/Users/mcdonadt/miniconda3/envs/sk050-biom215/lib/python3.5/site-packages/skbio/sequence/_grammared_sequence.py in _validate(self)
    358                    [str(b.tostring().decode("ascii")) for b in bad] if
    359                    len(bad) > 1 else bad[0],
--> 360                    list(self.alphabet)))
    361
    362     @stable(as_of='0.4.0')

ValueError: Invalid character in sequence: b'T'.
Valid characters: ['H', '-', 'A', 'N', 'M', 'G', 'R', 'Y', 'K', 'V', 'W', 'U', '.', 'S', 'C', 'B', 'D']
Note: Use `lowercase` if your sequence contains lowercase characters not in the sequence's alphabet.
```